### ISSUE TITLE
Enabling compression doesn't work

### ISSUE THREAD

**Issue by yamalight** [containous/traefik#2183]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Step 1: Started a new traefik instance using docker with enabled compression for a single http endpoint with the following config:
```toml
logLevel = "DEBUG"
debug = true
defaultEntryPoints = ["http"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
  compress = true

[docker]
  endpoint = "unix:///var/run/docker.sock"
```

Step 2: started simple web service inside docker that returned plain text content behind traefik
Step 3: used httpie / curl to get following headers:
```
HTTP/1.1 200 OK
Content-Length: 391
Content-Type: text/plain
Date: Thu, 28 Sep 2017 11:57:37 GMT
Vary: Accept-Encoding
```

Changing config to `compress = false` doesn't affect response in any way.

### What did you expect to see?

Expected to see gzip header and smaller content size

### What did you see instead?

No gzip header and original content size

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.3.8
Codename:     raclette
Go version:   go1.8.3
Built:        2017-09-07_08:46:19PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```json
{
 "GraceTimeOut": 10000000000,
 "Debug": true,
 "CheckNewVersion": true,
 "AccessLogsFile": "",
 "TraefikLogsFile": "",
 "LogLevel": "DEBUG",
 "EntryPoints": {
  "http": {
   "Network": "",
   "Address": ":80",
   "TLS": null,
   "Redirect": null,
   "Auth": null,
   "Compress": true
  }
 },
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [
  "http"
 ],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 180000000000,
 "InsecureSkipVerify": false,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "Docker": {
  "Watch": true,
  "Filename": "",
  "Constraints": null,
  "Endpoint": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "Domain": "",
  "TLS": null,
  "ExposedByDefault": true,
  "UseBindPortIP": false,
  "SwarmMode": false
 },
 "File": null,
 "Web": null,
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ConfigFile": "//traefik.toml"
}
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
INFO[2017-09-28T11:51:36Z] Traefik version v1.3.8 built on 2017-09-07_08:46:19PM
INFO[2017-09-28T11:51:36Z] Using TOML configuration file //traefik.toml
DEBU[2017-09-28T11:51:36Z] Global configuration loaded {"GraceTimeOut":10000000000,"Debug":true,"CheckNewVersion":true,"AccessLogsFile":"","TraefikLogsFile":"","LogLevel":"DEBUG","EntryPoints":{"http":{"Network":"","Address":":80","TLS":null,"Redirect":null,"Auth":null,"Compress":true}},"Cluster":null,"Constraints":[],"ACME":null,"DefaultEntryPoints":["http"],"ProvidersThrottleDuration":2000000000,"MaxIdleConnsPerHost":200,"IdleTimeout":180000000000,"InsecureSkipVerify":false,"Retry":null,"HealthCheck":{"Interval":30000000000},"Docker":{"Watch":true,"Filename":"","Constraints":null,"Endpoint":"unix:///var/run/docker.sock","Domain":"","TLS":null,"ExposedByDefault":true,"UseBindPortIP":false,"SwarmMode":false},"File":null,"Web":null,"Marathon":null,"Consul":null,"ConsulCatalog":null,"Etcd":null,"Zookeeper":null,"Boltdb":null,"Kubernetes":null,"Mesos":null,"Eureka":null,"ECS":null,"Rancher":null,"DynamoDB":null}
INFO[2017-09-28T11:51:36Z] Preparing server http &{Network: Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> Compress:true}
INFO[2017-09-28T11:51:36Z] Starting server on :80
INFO[2017-09-28T11:51:36Z] Starting provider *docker.Provider {"Watch":true,"Filename":"","Constraints":null,"Endpoint":"unix:///var/run/docker.sock","Domain":"","TLS":null,"ExposedByDefault":true,"UseBindPortIP":false,"SwarmMode":false}
DEBU[2017-09-28T11:51:36Z] Provider connection established with docker 17.06.2-ce (API 1.30)
DEBU[2017-09-28T11:51:36Z] Filtering container with empty frontend rule /hungry_lalande
DEBU[2017-09-28T11:51:36Z] Validation of load balancer method for backend backend-exo-yamalight-test-node-deploy failed: invalid load-balancing method ''. Using default method wrr.
DEBU[2017-09-28T11:51:36Z] Configuration received from provider docker: {"backends":{"backend-exo-yamalight-test-node-deploy":{"servers":{"server-exo-yamalight-test-node-deploy-0b54e7c0":{"url":"http://172.18.0.3:80","weight":0}},"loadBalancer":{"method":"wrr"}}},"frontends":{"frontend-Host-html-dev":{"entryPoints":["http"],"backend":"backend-exo-yamalight-test-node-deploy","routes":{"route-frontend-Host-html-dev":{"rule":"Host:html.dev"}},"passHostHeader":true,"priority":0,"basicAuth":[]}}}
DEBU[2017-09-28T11:51:36Z] Last docker config received more than 2s, OK
DEBU[2017-09-28T11:51:36Z] Creating frontend frontend-Host-html-dev
DEBU[2017-09-28T11:51:36Z] Wiring frontend frontend-Host-html-dev to entryPoint http
DEBU[2017-09-28T11:51:36Z] Creating route route-frontend-Host-html-dev Host:html.dev
DEBU[2017-09-28T11:51:36Z] Creating backend backend-exo-yamalight-test-node-deploy
DEBU[2017-09-28T11:51:36Z] Creating load-balancer wrr
DEBU[2017-09-28T11:51:36Z] Creating server server-exo-yamalight-test-node-deploy-0b54e7c0 at http://172.18.0.3:80 with weight 0
INFO[2017-09-28T11:51:36Z] Server configuration reloaded on :80
DEBU[2017-09-28T11:51:38Z] Round trip: http://172.18.0.3:80, code: 200, duration: 942.671µs
```

**Comment by errm** [comment]
The Compress middleware doesn't do anything until the Content-Length > 512 bytes, this is expected behaviour, @yamalight could you try with a larger response and confirm

**Comment by yamalight** [comment]
@errm yup, that works. thanks!
would be great to mention this in the docs IMO.

**Comment by traefiker** [comment]
Closed by #2184.
